### PR TITLE
dotCMS/core#24173 Create dialog for 'Create Page' button

### LINKED ISSUE #24173: UI - Favorite Page portlet: Create dialog for "Create Page" button
When clicked on "Create Page" button, then display a dialog with Page types content types and once you click on then you'll be redirected to create new content.
It should allow to filter pages content types

image

### PR DESCRIPTION
**Pull request by alfredo-dotcms**
### Proposed Changes
- Display a dialog with Page types content types and once you click on then you'll be redirected to create new content

https://user-images.githubusercontent.com/37185433/221702607-363e34b2-1a54-4f55-b2d5-f9cef394eac4.mov

### Checklist
- [ ] Tests
- [ ] Translations
- [ ] Security Implications Contemplated (add notes if applicable)

### Additional Info
** any additional useful context or info **

### Screenshots
Original             |  Updated
:-------------------------:|:-------------------------:
** original screenshot **  |  ** updated screenshot **

### CODE REVIEW

File: core-web/apps/dotcms-ui/src/app/portlets/dot-pages/dot-pages-create-page-dialog/dot-pages-create-page-dialog.component.html [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+        #searchInput
+        [placeholder]="'Search' | dm"
+        data-testid="dot-pages-create-page-dialog__keyword-input"
+        type="text"
+        pInputText
+        dotAutofocus
+    />
+</div>
+
+<div class="dot-pages-create-page-dialog__page-types-container" *ngIf="vm$ | async as vm">
```
Comment (hmoreras) [inline core-web/apps/dotcms-ui/src/app/portlets/dot-pages/dot-pages-create-page-dialog/dot-pages-create-page-dialog.component.html:13]: why no the `ngif` in all html ?

**Review by hmoreras — APPROVED** [review]

**Review by zulqarnainvd — APPROVED** [review]

File: core-web/apps/dotcms-ui/src/app/portlets/dot-pages/dot-pages.component.ts [hunk 2]
```diff
@@ -91,6 +94,18 @@ export class DotPagesComponent implements OnInit, OnDestroy {
...
 
+    /**
+     * Event to instantiate dialog with Create Page component
+     *
+     * @memberof DotPagesComponent
+     */
+    showCreatePageDialog(): void {
+        this.dialogService.open(DotPagesCreatePageDialogComponent, {
+            header: this.dotMessageService.get('create.page'),
+            width: '800px'
```
Comment (fmontes) [inline core-web/apps/dotcms-ui/src/app/portlets/dot-pages/dot-pages.component.ts:105]: Use rems.

**Review by fmontes — CHANGES_REQUESTED** [review]
.

File: core-web/apps/dotcms-ui/src/app/portlets/dot-pages/dot-pages-create-page-dialog/dot-pages-create-page-dialog.component.spec.ts [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+        fixture = TestBed.createComponent(DotPagesCreatePageDialogComponent);
+        de = fixture.debugElement;
+        dotRouterService = TestBed.inject(DotRouterService);
+        dialogRef = TestBed.inject(DynamicDialogRef);
+        fixture.detectChanges();
+    });
+
+    it('should have html components with attributes', () => {
+        expect(
+            de.query(By.css(`.dot-pages-create-page-dialog__search-container dot-icon`))
```
[the remark this answers is not in the document]
Comment (alfredo-dotcms) [reply core-web/apps/dotcms-ui/src/app/portlets/dot-pages/dot-pages-create-page-dialog/dot-pages-create-page-dialog.component.spec.ts]: fixed!

File: core-web/apps/dotcms-ui/src/app/portlets/dot-pages/dot-pages.component.ts (hunk 2 above)
Comment (alfredo-dotcms) [reply core-web/apps/dotcms-ui/src/app/portlets/dot-pages/dot-pages.component.ts]: fixed!

File: core-web/libs/data-access/src/lib/dot-es-content/dot-es-content.service.spec.ts [hunk 4]
```diff
@@ -39,7 +39,7 @@ describe('DotESContentService', () => {
         const req = httpMock.expectOne('/api/content/_search');
         expect(req.request.method).toBe('POST');
         expect(req.request.body).toBe(
-            '{"query":"  +deleted : false   +working : true   +contentType :  blog   +title : *  ","sort":"modDate DESC","limit":40,"offset":"0"}'
+            '{"query":"+contentType: blog   ","sort":"modDate DESC","limit":40,"offset":"0"}'
```
[the remark this answers is not in the document]
Comment (alfredo-dotcms) [reply core-web/libs/data-access/src/lib/dot-es-content/dot-es-content.service.spec.ts]: fixed!

File: core-web/libs/data-access/src/lib/dot-page-types/dot-page-types.service.ts [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+
+import { CoreWebService } from '@dotcms/dotcms-js';
+import { DotCMSContentType } from '@dotcms/dotcms-models';
+
+@Injectable()
+export class DotPageTypesService {
+    constructor(private coreWebService: CoreWebService) {}
+
+    /**
+     * Get the value of specific key
```
[the remark this answers is not in the document]
Comment (alfredo-dotcms) [reply core-web/libs/data-access/src/lib/dot-page-types/dot-page-types.service.ts]: fixed!

File: core-web/apps/dotcms-ui/src/app/portlets/dot-pages/dot-pages.module.ts [hunk 6]
```diff
@@ -37,12 +41,14 @@ import { DotPagesComponent } from './dot-pages.component';
...
         DotPageRenderService,
+        DotPageTypesService,
         DotTempFileUploadService,
         DotWorkflowsActionsService,
         DotWorkflowActionsFireService,
         DotWorkflowEventHandlerService,
         DotRouterService,
-        SiteService
+        SiteService,
+        DotPageStore
```
[the remark this answers is not in the document]
Comment (alfredo-dotcms) [reply core-web/apps/dotcms-ui/src/app/portlets/dot-pages/dot-pages.module.ts]: reverted!

File: core-web/apps/dotcms-ui/src/app/portlets/dot-pages/dot-pages-create-page-dialog/dot-pages-create-page-dialog.component.ts [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+})
+export class DotPagesCreatePageDialogComponent implements OnInit, OnDestroy {
+    @ViewChild('searchInput', { static: true }) searchInput: ElementRef;
+
+    vm$: Observable<DotPagesState> = this.store.vm$;
+
+    private destroy$: Subject<boolean> = new Subject<boolean>();
+
+    constructor(
+        private store: DotPageStore,
```
[the remark this answers is not in the document]
Comment (alfredo-dotcms) [reply core-web/apps/dotcms-ui/src/app/portlets/dot-pages/dot-pages-create-page-dialog/dot-pages-create-page-dialog.component.ts]: fixed, store removed and `DotPagesCreatePageDialogComponent` is a dummy components that gets data from inputs

**Review by fmontes — CHANGES_REQUESTED** [review]
Couple of comments.

**Review by fmontes — APPROVED** [review]

File: core-web/apps/dotcms-ui/src/app/portlets/dot-pages/dot-pages-listing-panel/dot-pages-listing-panel.component.ts [hunk 8]
```diff
@@ -15,6 +15,7 @@ import { DotActionsMenuEventParams } from '../dot-pages.component';
     styleUrls: ['./dot-pages-listing-panel.component.scss']
 })
 export class DotPagesListingPanelComponent {
+    @Output() createPage = new EventEmitter();
```
[the remark this answers is not in the document]
Comment (alfredo-dotcms) [reply core-web/apps/dotcms-ui/src/app/portlets/dot-pages/dot-pages-listing-panel/dot-pages-listing-panel.component.ts]: fixed

File: core-web/apps/dotcms-ui/src/app/portlets/dot-pages/dot-pages-create-page-dialog/dot-pages-create-page-dialog.component.ts [hunk 9]
```diff
@@ -0,0 +1,90 @@
...
+                if (target['value'].length) {
+                    this.pageTypes = this.pageTypesBackup.filter((pageType: DotCMSContentType) =>
+                        pageType.name
+                            .toLocaleLowerCase()
+                            .includes(target['value'].toLocaleLowerCase())
+                    );
+                } else {
+                    this.pageTypes = [...this.pageTypesBackup];
+                }
+            });
```
[the remark this answers is not in the document]
Comment (alfredo-dotcms) [reply core-web/apps/dotcms-ui/src/app/portlets/dot-pages/dot-pages-create-page-dialog/dot-pages-create-page-dialog.component.ts]: fixed

**Pull request merged**